engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -129,8 +130,8 @@ class EngineJobExecutionActor(replyTo: ActorRef,
       writeToMetadata(Map(callCachingReadResultMetadataKey -> "Cache Miss"))
       log.debug("Cache miss for job {}", jobTag)
       runJob(data)
-    case Event(hit @ CacheHit(cacheResultId), data: ResponsePendingData) =>
-      fetchCachedResults(data, jobDescriptorKey.call.task.outputs, hit)
+    case Event(hit: CacheHit, data: ResponsePendingData) =>
+      fetchCachedResults(jobDescriptorKey.call.task.outputs, hit.cacheResultIds.head) using data.withCacheHit(hit).popCacheHitId._2
```

Comment:
ToL: I personally prefer the `using` to go with the `goto`. I think that's why the `data` was passed in originally.

Also ToL: I'd prefer to have the data have some method that calculates `data.withCacheHit(hit).popCacheHitId._2` with a nice easy-to-understand name